cpp/src/arrow/compute/exec/options.h

```diff
@@ -361,6 +361,19 @@ class ARROW_EXPORT HashJoinNodeOptions : public ExecNodeOptions {
   Expression filter;
 };
 
+class ARROW_EXPORT AsofJoinNodeOptions : public ExecNodeOptions {
+ public:
+  AsofJoinNodeOptions(FieldRef time, FieldRef keys, int64_t tolerance)
+      : time(std::move(time)), keys(std::move(keys)), tolerance(tolerance) {}
+
+  // time column
+  FieldRef time;
+  // keys used for the join. All tables must have the same join key.
+  FieldRef keys;
```

Comment:
I agree. However, I'd prefer to leave as is since I will like to change this anyway when integrating with substrait (i.e. comments here https://github.com/apache/arrow/pull/13028/files#r882274989)